### PR TITLE
Add missing runtime dependencies in package.xml

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says.

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -18,9 +18,9 @@
   <buildtool_depend>ament_cmake_python</buildtool_depend>
   <buildtool_depend>python_cmake_module</buildtool_depend>
 
-  <build_depend>builtin_interfaces</build_depend>
-  <build_depend>rclcpp</build_depend>
-  <build_depend>rclpy</build_depend>
+  <depend>builtin_interfaces</depend>
```
Comment (clalancette) [inline package.xml:21]: As far as I can tell, `builtin_interfaces` is only ever used by the Python code in here.  Thus, I think it should be marked as an `exec_depend`, not a `depend`.

File: package.xml [hunk 2]
```diff
@@ -18,9 +18,9 @@
...
+  <depend>rclcpp</depend>
```
Comment (clalancette) [inline package.xml:22]: `rclcpp` isn't actually a `build_depend`, though it is required in the headers and in the tests.  So I think this should probably be:

```
<build_export_depend>rclcpp</build_export_depend>
<test_depend>rclcpp</test_depend>
```

File: package.xml [hunk 3]
```diff
@@ -18,9 +18,9 @@
...
+  <depend>rclpy</depend>
```
Comment (clalancette) [inline package.xml:23]: Similar to `builtin_interfaces`, `rclpy` is only ever used in the Python code, so it should be an `exec_depend`.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: package.xml (hunk 1 above)
Comment (hidmic) [reply package.xml:21]: You're absolutely right, which means the CMakeLists.txt is nonsense as well.

File: package.xml (hunk 2 above)
Comment (hidmic) [reply package.xml:22]: Agreed.

File: package.xml (hunk 3 above)
Comment (hidmic) [reply package.xml:23]: Indeed.

**Comment by hidmic** [comment]
@clalancette I'll close this PR and open another, more comprehensive review of this package's dependencies.

**Pull request closed without merging**